### PR TITLE
MAINTENANCE: rename sum to sum_labels and add a deprecation warning #123

### PR DESCRIPTION
**Pull request by tkoyama010**
MAINTENANCE: rename sum to sum_labels and add a deprecation warning #123

Join the big image sprint at SciPy Japan 2020 (#163)

### CODE REVIEW

File: dask_image/ndmeasure/__init__.py [hunk 1]
```diff
@@ -678,9 +678,9 @@ def standard_deviation(image, label_image=None, index=None):
     return std_lbl
 
 
-def sum(image, label_image=None, index=None):
+def sum_labels(image, label_image=None, index=None):
     """
-    Find the sum over an image at specified subregions.
+    Find the sum_labels over an image at specified subregions.
```
Comment (GenevieveBuckley) [inline dask_image/ndmeasure/__init__.py:683]:
```suggestion
    Find the sum of all pixels over specified subregions of an image.
```

**Comment by tkoyama010** [comment]
Thanks. It WFM. I will also add `DeprecationWarning` .

File: dask_image/ndmeasure/__init__.py [hunk 2]
```diff
@@ -29,7 +29,7 @@
     "minimum",
     "minimum_position",
     "standard_deviation",
-    "sum",
```
Comment (GenevieveBuckley) [inline dask_image/ndmeasure/__init__.py:32]: I think we will need to keep both `sum` and `sum_labels` in this list for now. We want people to still be able to use `sum`, but they will also see a warning to change their code to use `sum_labels` instead. Later on, we will remove it entirely.

File: tests/test_dask_image/test_ndmeasure/test_core.py [hunk 3]
```diff
@@ -2,7 +2,7 @@
 # -*- coding: utf-8 -*-
 
 from __future__ import absolute_import
-
+from distutils.version import LooseVersion
 import itertools as it
```
Comment (tkoyama010) [inline tests/test_dask_image/test_ndmeasure/test_core.py:6]:
```suggestion
import scipy
import itertools as it
```

File: dask_image/ndmeasure/__init__.py [hunk 4]
```diff
@@ -29,7 +29,7 @@
     "minimum",
     "minimum_position",
     "standard_deviation",
```
Comment (tkoyama010) [inline dask_image/ndmeasure/__init__.py:31]:
```suggestion
    "standard_deviation",
    "sum",
```

File: dask_image/ndmeasure/__init__.py (hunk 2 above)
Comment (tkoyama010) [reply dask_image/ndmeasure/__init__.py:32]: +1

**Comment by GenevieveBuckley** [comment]
Ok, now we need a`sum` function that will produce a deprecation warning and return the new `sum_labels` function instead.
Similar to this:
```python
def sum(image, label_image=None, index=None):
   """Docstring goes here"""
    warnings.warn("Deprecation message goes here", DeprecationWarning)
    return sum_labels(image, label_image=label_image, index=index)
```

File: tests/test_dask_image/test_ndmeasure/test_core.py [hunk 5]
```diff
@@ -110,7 +112,12 @@ def test_center_of_mass(datatype):
     ]
 )
 def test_measure_props(funcname, shape, chunks, has_lbls, ind):
-    sp_func = getattr(spnd, funcname)
+    if funcname == 'sum_labels' and scipy.__version__ < LooseVersion('1.5.0'):
```
Comment (tkoyama010) [inline tests/test_dask_image/test_ndmeasure/test_core.py:115]:
```suggestion
    if funcname == 'sum_labels' and scipy.__version__ > LooseVersion('1.5.0'):
```
It is my fault :cry:

File: tests/test_dask_image/test_ndmeasure/test_core.py [hunk 6]
```diff
@@ -110,7 +112,12 @@ def test_center_of_mass(datatype):
     ]
 )
 def test_measure_props(funcname, shape, chunks, has_lbls, ind):
-    sp_func = getattr(spnd, funcname)
+    if funcname == 'sum_labels' and scipy.__version__ > LooseVersion('1.5.0'):
```
Comment (GenevieveBuckley) [inline tests/test_dask_image/test_ndmeasure/test_core.py:115]:
```suggestion
    if funcname == 'sum_labels' and scipy.__version__ < LooseVersion('1.5.0'):
```

If scipy version is old, use function name `scipy.ndimage.sum`
But if scipy version is new, we can use the new `scipy.ndimage.sum_labels`

Comment (GenevieveBuckley) [reply tests/test_dask_image/test_ndmeasure/test_core.py:115]: Sorry if I have made this confusing. Please add a comment if you think that would make things less confusing for other people 😄

Comment (tkoyama010) [reply tests/test_dask_image/test_ndmeasure/test_core.py:115]: +1 for comment. I confused 😁 This is the place which we will need a comment 👍

File: dask_image/ndmeasure/__init__.py [hunk 7]
```diff
@@ -712,6 +713,12 @@ def sum(image, label_image=None, index=None):
     return sum_lbl
 
 
+def sum(image, label_image=None, index=None):
+    """DEPRECATED FUNCTION. Use $B!F(Bsum_labels$B!G(B instead."""
+    warnings.warn("DEPRECATED FUNCTION. Use $B!F(Bsum_labels$B!G(B instead.", DeprecationWarning)
```
Comment (tkoyama010) [inline dask_image/ndmeasure/__init__.py:718]:
```suggestion
    """DEPRECATED FUNCTION. Use `sum_labels` instead."""
    warnings.warn("DEPRECATED FUNCTION. Use `sum_labels` instead.", DeprecationWarning)
```

**Review by tkoyama010 — COMMENTED** [review]
LGTM

**Review by GenevieveBuckley — APPROVED** [review]
Looks good to me! Excellent work @tkoyama010

**Comment by tkoyama010** [comment]
@GenevieveBuckley Thanks for great help ☺

**Pull request merged**